test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -733,11 +733,15 @@ public void testEqualsAndHashcode() {
         }
     }
 
+    /**
+     * Generic test that checks that the toString method renders the XContent
+     * correctly.
+     */
     public void testToString() throws IOException {
         for (int runs = 0; runs < NUMBER_OF_TESTQUERIES; runs++) {
             QB testQuery = createTestQueryBuilder();
             XContentType xContentType = XContentType.JSON;
-            String toString = Strings.toString(testQuery);
+            String toString = randomBoolean() ? Strings.toString(testQuery) : testQuery.toString();
```

Comment:
I don't think this should be randomized. If we test toString, we ought to test that one all the time. If we also want to test something else, I think we should make that part a separate test method.